oasislmf/pytools/aal/manager.py

```diff
@@ -901,20 +918,48 @@ def run(run_dir, subfolder, aal_output_file=None, alct_output_file=None, meanonl
                     sample_size,
                     file_data["no_of_periods"],
                 )
-            write_ndarray_to_fmt_csv(output_files["aal"], np.array(aal_data, dtype=AAL_dtype), AAL_headers, AAL_fmt)
-        if output_alct:
+            aal_data = np.array(aal_data, dtype=outmap["aal"]["dtype"])
```

Comment:
One suggestion to avoid those ifs
in the init phase create a write function for each output format that take out_data and outmap[<type>] and do the writing.
for example for parquet:
```
def write_out(out_data, out_type):
    data_df = pd.DataFrame(out_data)
    data_table = pa.Table.from_pandas(data_df)
    out_type["file"].write_table(data_table)
```
then to write you just call write_out(aal_data, outmap["aal"])